web_ui/frontend/app/registry/page.tsx

```diff
@@ -18,29 +18,112 @@
+            })
+            data = responseData
+        }
+
+        return data
+    }
+
+    const _setData = async () => {setData(await getData())}
+
+    useEffect(() => {
+        _setData();
+        (async () => {
+            if(await isLoggedIn()){
+                setAuthenticated(getAuthenticated() as Authenticated)
+            }
+        })();
+    }, [])
+
+    const pendingData = useMemo(
+        () => data?.filter(
+            (namespace) => namespace.admin_metadata.status === "Pending" &&
+                (authenticated?.user == namespace.admin_metadata.user_id || authenticated?.role == "admin")
+                ), [data, authenticated]
+    )
+    const approvedCacheData = useMemo(
+        () => data?.filter(
+            (namespace) => namespace.admin_metadata.status === "Approved" && namespace.type == "cache"
+        ),
+        [data]
+    )
+    const approvedOriginData = useMemo(
+        () => data?.filter(
+            (namespace) => namespace.admin_metadata.status === "Approved" && namespace.type == "origin"
+        ),
+        [data]
+    )
+
     return (
         <Box width={"100%"}>
             <Grid container spacing={2}>
-                <Grid item xs={12} lg={6}>
-                    <Typography variant={"h4"}>Origins</Typography>
-                    <NamespaceTable type={"origin"}/>
+                <Grid item xs={12}>
+                    <Typography variant={"h4"}>Namespace Registry</Typography>
+                    <Collapse in={alert !== undefined}>
+                        <Box mt={2}>
+                            <Alert severity={alert?.severity}>{alert?.message}</Alert>
+                        </Box>
+                    </Collapse>
+                </Grid>
+                <Grid item xs={12} lg={8} justifyContent={"space-between"}>
+                    <UnauthenticatedContent pb={2}>
+                        <Typography variant={"body1"}>
+                            Login to register new namespaces.
+                            <Button sx={{ml:2}} variant={"contained"} size={"small"} color={"primary"} href={"/view/login/index.html"}>Login</Button>
+                        </Typography>
+                    </UnauthenticatedContent>
+                    {
+                        pendingData && pendingData.length > 0 &&
+                        <Grid item xs={12}>
+                            <Typography variant={"h5"} pb={2}>Pending Registrations</Typography>
+
+                            {pendingData.map((namespace) => <PendingCard key={namespace.id} namespace={namespace} authenticated={authenticated} onAlert={(a) => setAlert(a)} onUpdate={_setData}/>)}
+                        </Grid>
+                    }
+
+                    <Typography variant={"h5"} py={2}>Origins</Typography>
+                    { approvedOriginData !== undefined ? approvedOriginData.map((namespace) => <Card key={namespace.id} namespace={namespace} authenticated={authenticated}/>) : <NamespaceCardSkeleton/> }
```

Comment:
There seems to be no way to check the detail of a registration unless to edit it? Yes?

Reply:
If you click on the bar it will drop down with the namespace details.

Reply:
You are right. I though that's only admin_metadata but we don't need to see `pubkey` and `identity` so I guess that's all the details we want